apps/project/tests/test_mutations.py

```diff
@@ -1111,3 +1111,67 @@ def test_user_group_membership_using_clairvoyan_one_bulk(self):
 
     def test_user_group_membership_admin_bulk(self):
         self._user_group_membership_bulk(self.project_role_admin)
+
+    def test_project_deletion(self):
+        query = '''
+            mutation MyMutation($projectId: ID!) {
+              __typename
+              project(id: $projectId) {
+                  projectDelete {
+                    ok
+                    errors
+                    result {
+                      id
+                      title
+                      isDeleted
+                    }
+                  }
+              }
+            }
+        '''
+
+        admin_user = UserFactory.create()
+        member_user = UserFactory.create()
+        af = AnalysisFrameworkFactory.create()
+        project = ProjectFactory.create(analysis_framework=af)
+
+        project.add_member(admin_user, role=self.project_role_admin)
+        project.add_member(member_user, role=self.project_role_member)
+
+        def _query_check(**kwargs):
+            return self.query_check(
+                query,
+                mnested=['project'],
+                variables={'projectId': project.id},
+                **kwargs,
+            )
+        # without login
+        _query_check(assert_for_error=True)
+        # ---------- With login and member_user in project
+        self.force_login(member_user)
+        _query_check(okay=False)
+
+        # ------ Login with admin_user
+        self.force_login(admin_user)
+        response = _query_check(okay=True)
+        # make sure that the `is_deleted` field is set to True
+        self.assertEqual(response['data']['project']['projectDelete']['result']['isDeleted'], True)
+
+    def test_project_deletion_celery_task(self):
+        old_project_count = Project.objects.count()
+        admin_user = UserFactory.create()
+        af = AnalysisFrameworkFactory.create()
+        project = ProjectFactory.create(analysis_framework=af)
+
+        project.add_member(admin_user, role=self.project_role_admin)
+        # check for project_count
+        self.assertEqual(Project.objects.count(), old_project_count + 1)
+        # now delete the project
+        project.is_deleted = True
+        project.deleted_at = '2022-01-01'
+        project.save()
+
+        # call the deletion method
+        project_deletion()
+
+        self.assertEqual(Project.objects.count(), old_project_count)
```

Comment:
Let's add more checks. Create multiple projects w/wo delete tagged. And use timezone mock to check if it deletes in correct time. Also use ids instead of count for checking if project exists or not.

Also, admin_user is note required here.